packages/expo-dev-menu/android/src/debug/java/expo/modules/devmenu/DevMenuManager.kt

```diff
@@ -207,6 +210,11 @@ object DevMenuManager : DevMenuManagerInterface, LifecycleEventListener {
       ?: reactContext.applicationContext as Application)
     maybeStartDetectors(devMenuHost.getContext())
 
+    settings = testInterceptor.overrideSettings()
+    if (settings != null) {
+      return
```

Comment:
I'm leery of this `return` because it would be very easy to add more logic to this method below that is intended to run in every case (including in tests where we're overriding `settings`). Could we do something like this instead:

```
val overriddenSettings = testInterceptor.overrideSettings()
settings = if (overriddenSettings != null) {
  settings = overriddenSettings
} else if (reactContext.hasNativeModule...
```